Function anyTrue and compilation

The compilation of a .mo file fails if it includes a block _Modelica.Blocks.MathBoolean.Or_, which involves the function **anyTrue** in order to process a boolean vector.

I guess the compilation would fail too with other blocks from MathBoolean.

The message from the compilation log :

Scanning dependencies of target lib
[ 16%] Building CXX object CMakeFiles/lib.dir/GeneratorSynchronousFourWindingsGoverGt2VRSemipol2Tfo.cpp.o
In file included from /home/guicharderw/Projects/dynawo/dynawo-rte/build/gcc9/460_add_mteret/Release/dynawo-rte/sources/Models/Modelica/P/GeneratorSynchronousFourWindingsGoverGt2VRSemipol2Tfo/GeneratorSynchronousFourWindingsGoverGt2VRSemipol2Tfo.cpp:4:
/home/guicharderw/Projects/dynawo/dynawo-rte/build/gcc9/460_add_mteret/Release/dynawo-rte/sources/Models/Modelica/P/GeneratorSynchronousFourWindingsGoverGt2VRSemipol2Tfo/GeneratorSynchronousFourWindingsGoverGt2VRSemipol2Tfo_Dyn.h:172:65: error: ‘boolean_array’ has not been declared
  172 |      modelica_boolean omc_Modelica_Math_BooleanVectors_anyTrue( boolean_array _b);

--- a/dynawo/sources/Modeler/ModelManager/DYNModelManagerOwnFunctions.cpp
+++ b/dynawo/sources/Modeler/ModelManager/DYNModelManagerOwnFunctions.cpp
@@ -879,6 +879,43 @@ _index_t* integer_array_make_index_array(const integer_array_t *arr) {
     return reinterpret_cast<int*>(arr->data);
 }
 
+/* allocates n booleans in the boolean_buffer */
+m_boolean* boolean_alloc(int n) {
+  return reinterpret_cast<m_boolean*>(malloc(n*sizeof(m_boolean)));
+}
+
+m_boolean* boolean_array_element_addr1(const boolean_array_t* source, int /*ndims*/, int dim1) {
+    return boolean_ptrget(source, dim1 - 1);
+}
+
+void simple_alloc_1d_boolean_array(boolean_array_t* dest, int n) {
+    simple_alloc_1d_base_array(dest, n, boolean_alloc(n));
+}
+
+
+void put_boolean_element(m_boolean value, int i1, boolean_array_t *dest) {
+    /* Assert that dest has correct dimension */
+    /* Assert that i1 is a valid index */
+    boolean_set(dest, i1, value);
+}
+
+
+/* array_alloc_scalar_boolean_array
+ *
+ * Creates(incl allocation) an array from scalar elements.
+ */
+
+void array_alloc_scalar_boolean_array(boolean_array_t* dest, int n, ...) {
+    int i;
+    va_list ap;
+    simple_alloc_1d_boolean_array(dest, n);
+    va_start(ap, n);
+    for (i = 0; i < n; ++i) {
+        put_boolean_element((m_boolean) va_arg(ap, int), i, dest);
+    }
+    va_end(ap);
+}
+
 /* Fills an array with a value. */
 void fill_alloc_real_array(real_array_t* dest, modelica_real value, int ndims, ...) {
     size_t i;

--- a/dynawo/sources/Modeler/ModelManager/DYNModelManagerOwnFunctions.h
+++ b/dynawo/sources/Modeler/ModelManager/DYNModelManagerOwnFunctions.h
@@ -269,6 +269,52 @@ void create_integer_array_from_range(integer_array_t *dest, modelica_integer sta
 _index_t* integer_array_make_index_array(const integer_array_t *arr);
 
 /**
+ * Method copied from <OpenModelica Sources>/SimulationRuntime/c/gc/memory_pool.h
+ * It is needed for Dynawo models dynamic libraries compilation
+ */
+m_boolean* boolean_alloc(int n);
+
+/**
+ * Method copied from <OpenModelica Sources>/SimulationRuntime/c/util/boolean_array.h
+ * It is needed for Dynawo models dynamic libraries compilation
+ */
+inline modelica_boolean *boolean_ptrget(const boolean_array_t *a, size_t i) {
+    return reinterpret_cast<modelica_boolean *>(a->data) + i;
+}
+
+/**
+ * Method copied from <OpenModelica Sources>/SimulationRuntime/c/util/boolean_array.h
+ * It is needed for Dynawo models dynamic libraries compilation
+ */
+m_boolean* boolean_array_element_addr1(const boolean_array_t* source, int ndims, int dim1);
+
+/**
+ * Method copied from <OpenModelica Sources>/SimulationRuntime/c/util/boolean_array.h
+ * It is needed for Dynawo models dynamic libraries compilation
+ */
+inline void boolean_set(boolean_array_t *a, size_t i, modelica_boolean r) {
+    reinterpret_cast<modelica_boolean *> (a->data)[i] = r;
+}
+
+/**
+ * Method copied from <OpenModelica Sources>/SimulationRuntime/c/util/boolean_array.h
+ * It is needed for Dynawo models dynamic libraries compilation
+ */
+extern void simple_alloc_1d_boolean_array(boolean_array_t* dest, int n);
+
+/**
+ * Method copied from <OpenModelica Sources>/SimulationRuntime/c/util/boolean_array.h
+ * It is needed for Dynawo models dynamic libraries compilation
+ */
+void put_boolean_element(m_boolean value, int i1, boolean_array_t* dest);
+
+/**
+ * Method copied from <OpenModelica Sources>/SimulationRuntime/c/util/boolean_array.h
+ * It is needed for Dynawo models dynamic libraries compilation
+ */
+void array_alloc_scalar_boolean_array(boolean_array_t* dest, int n, ...);
+
+/**
  * Method copied from <OpenModelica Sources>/SimulationRuntime/c/util/real_array.h
  * It is needed for Dynawo models dynamic libraries compilation
  */

--- a/dynawo/sources/Modeler/ModelManager/DYNModelManagerOwnTypes.h
+++ b/dynawo/sources/Modeler/ModelManager/DYNModelManagerOwnTypes.h
@@ -77,6 +77,9 @@ typedef integer_array_t integer_array;
 typedef real_array_t real_array;
 typedef string_array_t string_array;
 
+typedef base_array_t boolean_array_t;
+typedef boolean_array_t boolean_array;
+
 #endif  // DOXYGEN_SHOULD_SKIP_THIS
 
 #endif  // MODELER_MODELMANAGER_DYNMODELMANAGEROWNTYPES_H_

--- a/dynawo/sources/ModelicaCompiler/Scripts_OMC_1_13_2/utils.py
+++ b/dynawo/sources/ModelicaCompiler/Scripts_OMC_1_13_2/utils.py
@@ -975,7 +975,7 @@ def convert_booleans_body (boolean_variables_names, body):
 # @param line : the line to convert
 # @return the converted line as a string
 def convert_booleans_line (boolean_variables_names, line):
-    allowed_containers = [' ', '(', ')', '=', '!']
+    allowed_containers = [' ', '(', ')', '=', '!', ',']
     # temporary patterns for avoiding to convert quoted names
     patterns_temp = ["@@@@@@@@", "$$$$$$$$"]
     for n, pattern in enumerate(patterns_temp):

